src/HDF5.jl

```diff
@@ -158,117 +158,117 @@ function show(io::IO, fid::HDF5File)
     if isvalid(dset)
         print(io, "HDF5 dataset: ", name(dset), " (file: ", dset.file.filename, " xfer_mode: ", dset.xfer.id, ")")
     else
         print(io, "HDF5 dataset (invalid)")
     end
 end
 
-mutable struct HDF5Datatype
+mutable struct Datatype
     id::hid_t
     toclose::Bool
-    file::HDF5File
+    file::File
 
-    function HDF5Datatype(id, toclose::Bool=true)
+    function Datatype(id, toclose::Bool=true)
         nt = new(id, toclose)
         if toclose
             finalizer(close, nt)
         end
         nt
     end
-    function HDF5Datatype(id, file::HDF5File, toclose::Bool=true)
+    function Datatype(id, file::File, toclose::Bool=true)
         nt = new(id, toclose, file)
         if toclose
             finalizer(close, nt)
         end
         nt
     end
 end
-convert(::Type{hid_t}, dtype::HDF5Datatype) = dtype.id
-show(io::IO, dtype::HDF5Datatype) = print(io, "HDF5 datatype: ", h5lt_dtype_to_text(dtype.id))
-hash(dtype::HDF5Datatype, h::UInt) =
+convert(::Type{hid_t}, dtype::Datatype) = dtype.id
+show(io::IO, dtype::Datatype) = print(io, "HDF5 datatype: ", h5lt_dtype_to_text(dtype.id))
+hash(dtype::Datatype, h::UInt) =
     (dtype.id % UInt + h) ^ (0xadaf9b66bc962084 % UInt)
-==(dt1::HDF5Datatype, dt2::HDF5Datatype) = h5t_equal(dt1, dt2) > 0
+==(dt1::Datatype, dt2::Datatype) = h5t_equal(dt1, dt2) > 0
 
 # Define an H5O Object type
-const HDF5Object = Union{HDF5Group, HDF5Dataset, HDF5Datatype}
+const Object = Union{Group, Dataset, Datatype}
 
-mutable struct HDF5Dataspace
+mutable struct Dataspace
     id::hid_t
 
-    function HDF5Dataspace(id)
+    function Dataspace(id)
         dspace = new(id)
         finalizer(close, dspace)
         dspace
     end
 end
-convert(::Type{hid_t}, dspace::HDF5Dataspace) = dspace.id
+convert(::Type{hid_t}, dspace::Dataspace) = dspace.id
 
-mutable struct HDF5Attribute
+mutable struct Attribute
     id::hid_t
-    file::HDF5File
+    file::File
 
-    function HDF5Attribute(id, file)
+    function Attribute(id, file)
         dset = new(id, file)
         finalizer(close, dset)
         dset
     end
 end
-convert(::Type{hid_t}, attr::HDF5Attribute) = attr.id
-show(io::IO, attr::HDF5Attribute) = isvalid(attr) ? print(io, "HDF5 attribute: ", name(attr)) : print(io, "HDF5 attribute (invalid)")
+convert(::Type{hid_t}, attr::Attribute) = attr.id
+show(io::IO, attr::Attribute) = isvalid(attr) ? print(io, "HDF5 attribute: ", name(attr)) : print(io, "HDF5 attribute (invalid)")
 
-mutable struct HDF5Attributes
-    parent::Union{HDF5File, HDF5Group, HDF5Dataset}
+mutable struct Attributes
+    parent::Union{File, Group, Dataset}
 end
-attrs(p::Union{HDF5File, HDF5Group, HDF5Dataset}) = HDF5Attributes(p)
+attrs(p::Union{File, Group, Dataset}) = Attributes(p)
 
 # Methods for reference types
-function HDF5ReferenceObj(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, name::String)
+function HDF5ReferenceObj(parent::Union{File, Group, Dataset}, name::String)
```

Comment:
Shouldn't `HDF5ReferenceObj` be renamed as well? The fact that v1.10 has two different types of references makes this less nice, but since we also don't support he dataset region references in the high-level API, this could just be `Reference`. Then once the library is upgraded to v1.12, the underlying storage type can be changed to `H5R_ref_t` which will support both of the existing object and dataset region reference types and then gets the new attribute reference for free.

Reply:
Agreed